pkg/apps/application-list.jsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import PackageKit from "./packagekit";
+import { left_click, icon_url, show_error, launch, ProgressBar, CancelButton } from  "./utils.jsx";
+
+const _ = cockpit.gettext;
+
+class ApplicationRow extends React.Component {
+    constructor() {
+        super();
+        this.state = { progress: false };
```

Comment:
Shouldn't this initialize `progress_title` too? Also, since `progress` seems to be some percentage, a default value of -1 or `null` might be better - `false` indicates that it's (always) a boolean value.

Reply:
`progress_title` only matters when `progress` is not `false`.  I'll use `-1` to denote the "idle" state.